assets/src/infrastructure/ui/display/AlertDialog.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import {
+	AlertDialog as ChakraAlertDialog,
+	AlertDialogBody,
+	AlertDialogFooter,
+	AlertDialogHeader,
+	AlertDialogContent,
+	AlertDialogOverlay,
+	Button,
+} from '@chakra-ui/core';
+import { __ } from '@wordpress/i18n';
+
+import { EspressoButton } from '@application/ui/input';
+import { AlertDialogProps } from './types';
+
+const AlertDialog: React.FC<AlertDialogProps> = ({ buttonProps, onConfirm, title }) => {
+	const [isOpen, setIsOpen] = React.useState(false);
+	const cancelRef = React.useRef();
+	const onClose = () => setIsOpen(false);
+	const onDelete = () => {
+		onClose();
+		onConfirm();
+	};
```

Comment:
replaced the order, but in this case since `onConfirm` is not an async operation, both will be fired at almost the same time
***
in order to execute `onClose` after `onConfirm`, the last needs to returns a promise